en/debug/eclipse_jlink.md

```diff
@@ -0,0 +1,83 @@
+   - On Linux the tools are installed in **usr/bin**.
+
+For more information, see: https://gnu-mcu-eclipse.github.io/debug/jlink/install/
+
+## First Use
+
+1. Run *Eclipse*.
+1. Add a source by choosing **File > Import > C/C++ > Existing Code as Makefile Project** and click **Next**.
+1. Point it to the Firmware folder and give it a name, then select *ARM Cross GCC* in the *Toolchain for Indexer Settings* and click **Finish**.
+   Import takes a while, wait for it to finish
+1. Set the MCU settings: right-click on the top-level project in the Project Explorer, select “Properties” then under MCU choose SEGGER J-Link Path. Set it like the screenshot below shows.
+   ![Eclipse: Segger J-Link Path](../../assets/debug/eclipse_segger_jlink_path.png)
+1. Update packs: 
+   - Click the small icon on the top right called *Open Perspective* and open the “Packs” perspective.
+   ![Eclipse: Workspace](../../assets/debug/eclipse_workspace_perspective.png)
+   - Click the update all button.
+     > **Tip** This takes a VERY LONG TIME (10 minutes). Ignore all the errors about missing packages that pop up.
+     
+     ![Eclipse: Workspace Packs Perspective](../../assets/debug/eclipse_packs_perspective.jpg)
+   - The STM32Fxx devices are found in the Keil folder, install by right-clicking and then selecting **install** on the according device for F4 and F7.
+1. Setup debug configuration for target: 
+   - Right click project and open the *Settings* (menu: **C/C++ Build > Settings**)
+   - Choose the *Devices* Tab, *Devices* section (Not *Boards*).
+   - Find the FMU chip you wish to debug.
+   
+   ![Eclipse: Select FMU in settings](../../assets/debug/eclipse_settings_devices_fmu.png)
+1. Select debug configurations with the small drop-down next to the bug symbol
+   ![Eclipse: Debug config](../../assets/debug/eclipse_settings_debug_config.png)
+1. Then select *GDB SEGGER J-Link Debugging* and then the **New config** button on the top left.
+   ![Eclipse: GDB Segger Debug config](../../assets/debug/eclipse_settings_debug_config_gdb_segger.png)
+1. Setup build config:
+   - Give it a name and set  the *C/C++ Application* to the corresponding **.elf** file.
+   - Choose *Disable Auto build*
+     > **Note** Remember that you must build the target from the command line before starting a debug session.
+     
+   ![Eclipse: GDB Segger Debug config](../../assets/debug/eclipse_settings_debug_config_gdb_segger_build_config.png)
+1. The *Debugger* and *Startup* tabs shouldn’t need any modifications (just verify your settings with the screenshots below)
+
+   ![Eclipse: GDB Segger Debug config: debugger tab](../../assets/debug/eclipse_settings_debug_config_gdb_segger_build_config_debugger_tab.png)
+   ![Eclipse: GDB Segger Debug config: startup tab](../../assets/debug/eclipse_settings_debug_config_gdb_segger_build_config_startup_tab.png)
+
+
+## Appendix
```

Comment:
When do we need to do this stuff in the appending? Ie in what of the step above does this become important? If it is always important, should it be embedded in those steps?

Reply:
the stuff in the appendix is the more manual way to get the register view working. You need this when you can not find the target cpu in the package manager. for me, this was the case for stm f7 controller. do you think it deserves its own chapter?

Reply:
Sort of. I added it to a troubleshooting section.